modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -2223,6 +2226,16 @@ public static synchronized boolean isLocalHostChanged() throws IOException {
                     }
                 }
 
+                try {
+                    InetAddress localHost = InetAddress.getByName(LOCALHOST);
+
+                    if (!locAddrs.contains(localHost))
+                        locAddrs.add(localHost);
+                }
+                catch (Exception ex) {
+                    // Ignore.
+                }
+
```

Comment:
What's the purpose of this code? What is the scenario which doesn't work without it?

Reply:
Let me add a comment about that.